react/components/InfoCard/index.js

```diff
@@ -168,6 +176,32 @@ const InfoCard = ({
 
   const subheadClasses = `${handles.infoCardSubhead} t-body mt6 c-on-base ${alignToken} mw-100`
 
+  const infoCardRef = useRef(null)
+  
+  const { push } = usePixel()
+
+  const promotionEventData =
```

Comment:
This seems to be used only inside the onView function, what do you think to put it there?

Reply:
Hi @igorbrasileiro! Thanks for review my PR.

Right! I changed the localization of promotionEventData to inside the useOnView. I'm not sure if this are your request too or only the promotionEventData, but hooks aren't must localize in the body of the component? I can't changed this to inside the useOnView.

Reply:
Hi @matheusmunarao!
I was about to leave the same suggestion as @igorbrasileiro 😅 .

I think what he meant is that this variable `promotionEventData` is only being used inside of the `onView` callback you're passing to the `useOnView` hook. Hence this variable doesn't need to be in the component's body but could be moved inside of the callback.